src/Compilers/Core/Portable/Emit/EditAndContinue/DeltaMetadataWriter.cs

```diff
@@ -539,60 +539,75 @@ protected override void CreateIndicesForNonTypeMembers(ITypeDefinition typeDef)
 
             int typeRowId = _typeDefs.GetRowId(typeDef);
 
+            // First we find the deleted methods, and add them to our dictionary. This is used later when
+            // processing events, properties, methods, and references.
+            var deletedMethods = _changes.GetDeletedMethods(typeDef);
+            if (deletedMethods.Length > 0)
+            {
+                var deletedTypeMembers = ImmutableDictionary.CreateBuilder<IMethodDefinition, DeletedMethodDefinition>(ReferenceEqualityComparer.Instance);
+                foreach (var methodDef in deletedMethods)
+                {
+                    var oldMethodDef = (IMethodDefinition)methodDef.GetCciAdapter();
+                    deletedTypeMembers.Add(oldMethodDef, new DeletedMethodDefinition(oldMethodDef, typeDef, _typesUsedByDeletedMembers));
+                }
+
+                _deletedTypeMembers.Add(typeDef, deletedTypeMembers.ToImmutableDictionary());
+            }
+
             foreach (var eventDef in typeDef.GetEvents(this.Context))
             {
                 if (!_eventMap.Contains(typeRowId))
                 {
                     _eventMap.Add(typeRowId);
                 }
 
-                this.AddDefIfNecessary(_eventDefs, eventDef);
+                var eventChange = _changes.GetChange(eventDef);
+                eventChange = _changes.FixSymbolChangeForReAddedMembers(eventDef, eventChange, DefinitionExistsInAnyPreviousGeneration);
```

Comment:
>DefinitionExistsInAnyPreviousGeneration [](http://example.com/codeflow?start=95&length=39)

Maybe call this upfront and pass a boolean? It's just a couple of dictionary lookups so we are not saving much by sometimes not calling this. Also if we call it upfront we can remove the type switch since now we know what kind of member are we dealing with

Reply:
IMO that would make things messy. The problem is that `FixSymbolChangeForReAddedMembers` is [a little bit recursive](https://github.com/dotnet/roslyn/pull/63486/files#diff-046e818f68863c0df20880468e530e28bb53a00ae97ca49ee6951275464538edR399) so we'd have to bring some of that logic out to here. Personally I like that its all in that one method.

Reply:
I see, did not noticed that.